tests/functional/http/test_providers.py

```diff
@@ -21,24 +39,55 @@ def test_send_when_web3_error_raises_transaction_error(
         )
         provider._web3 = mock_web3
 
-        class MockReceipt(ReceiptAPI):
-            @classmethod
-            def decode(cls, data: dict) -> "ReceiptAPI":
-                return MockReceipt(
-                    txn_hash="test-hash",  # type: ignore
-                    status=TransactionStatusEnum.NO_ERROR,  # type: ignore
-                    gas_used=0,  # type: ignore
-                    gas_price="60000000000",  # type: ignore
-                    block_number=0,  # type: ignore
-                )
-
-        mock_network_api.ecosystem.receipt_class = MockReceipt
+        mock_network_api.ecosystem.receipt_class = _create_mock_receipt()
         web3_error_data = {
             "code": -32000,
-            "message": "Unable to perform actual",
+            "message": "Test Error Message",
         }
         mock_web3.eth.send_raw_transaction.side_effect = ValueError(web3_error_data)
-        with pytest.raises(VirtualMachineError) as err:
+        with pytest.raises(TransactionError) as err:
             provider.send_transaction(mock_transaction)
 
         assert web3_error_data["message"] in str(err.value)
+
+    def test_send_transactions_failing_and_used_all_gas_raises_out_of_gas_error(
```

Comment:
lol this method name

Reply:
It is a test!

`test_<condition>_<expectation>()`

Reply:
I trimmed it down some